lib/src/console/command/set_command.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:cider/src/console/command/application_command.dart';
+import 'package:cider/src/console/console.dart';
+import 'package:cider/src/console/exit_code.dart';
+import 'package:pub_semver/pub_semver.dart';
+
+class SetCommand extends ApplicationCommand {
+  SetCommand(this._console);
+
+  final Console _console;
+
+  @override
+  final description = 'Setting given version explicitly';
+
+  @override
+  final name = 'set';
+
+  @override
+  final aliases = ['switch', 'change'];
+
+  @override
+  int run() {
+    try {
+      if (argResults.rest.isEmpty) {
+        throw UnsupportedError('Version must be specified');
+      } else if (argResults.rest.length > 1) {
+        throw UnsupportedError('Too many arguments passed');
+      } else {
+        Version version;
+        try {
+          version = Version.parse(argResults.rest.single);
+        } on dynamic catch (_) {
+          throw UnsupportedError('Can\'t parse version');
+        }
+        _console.log(createApp().set(version));
+      }
+    } on UnsupportedError catch (e) {
+      _console.error(e?.message ?? 'An unsupported error occurred');
+      return ExitCode.usageException;
+    } on dynamic catch (e) {
```

Comment:
I'm not sure I understand the expected behavior here. Is there a supporting test?

Reply:
Not all UnsupportedError have 'message'. Nullable field.

```
/**
 * The operation was not allowed by the object.
 *
 * This [Error] is thrown when an instance cannot implement one of the methods
 * in its signature.
 */
@pragma("vm:entry-point")
class UnsupportedError extends Error {
  final String? message; // <= Nullable field.
  @pragma("vm:entry-point")
  UnsupportedError(String this.message);
  String toString() => "Unsupported operation: $message";
}
```

Reply:
Still unclear what it has to do with `} on dynamic catch (e) {`. Can you elaborate please?